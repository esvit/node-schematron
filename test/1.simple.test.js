const { getSchematronRequest, getSchematronResults, parseDom } = require('./util');

// Source of XML and Schematron samples:
// https://www.data2type.de/en/xml-xslt-xslfo/schematron/schematron-reference/

describe('1. Simplest Schematron validation', () => {
	const { patterns: [ { rules: [ firstRule ] } ] } = (schematronRequest = getSchematronRequest(
		parseDom(`
			<schema xmlns="http://purl.oclc.org/dsdl/schematron">
				<pattern>
					<rule context="//animal">
						<report test="species = 'lion' and age &gt; 3">An adult lion</report>
						<report test="species = 'elephant' and age &gt; 18">An adult elephant</report>
						<assert test="age &lt; 999">The animal is alive</assert>
					</rule>
				</pattern>
			</schema>
		`)
	));

	it('getSchematronRequest', () => expect(schematronRequest).toMatchSnapshot());

	const [ [ [ lionResult, elephantResult ] ] ] = (schematronResults = getSchematronResults(
		schematronRequest,
		parseDom(`
			<room>
				<animal sex="female" carnivore="yes">
					<species>lion</species>
					<weight>200</weight>
					<age>10</age>
				</animal>
				<animal sex="male" carnivore="no">
					<species>elephant</species>
					<weight>3000</weight>
					<age>21</age>
				</animal>
			</room>
		`)
	));

	it('getSchematronResults', () => {
		expect(lionResult).toMatchSnapshot();
		expect(elephantResult).toMatchSnapshot();
	});
});
