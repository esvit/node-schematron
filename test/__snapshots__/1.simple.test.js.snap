// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. Simplest Schematron validation getSchematronRequest 1`] = `
Object {
  "patterns": Array [
    Object {
      "id": null,
      "rules": Array [
        Object {
          "context": "//animal",
          "tests": Array [
            Object {
              "isAssertion": false,
              "isReport": true,
              "message": Array [
                "An adult lion",
              ],
              "test": "species = 'lion' and age > 3",
            },
            Object {
              "isAssertion": false,
              "isReport": true,
              "message": Array [
                "An adult elephant",
              ],
              "test": "species = 'elephant' and age > 18",
            },
            Object {
              "isAssertion": true,
              "isReport": false,
              "message": Array [
                "The animal is alive",
              ],
              "test": "age < 999",
            },
          ],
          "variables": Array [],
        },
      ],
    },
  ],
  "title": null,
}
`;

exports[`1. Simplest Schematron validation getSchematronResults 1`] = `
Object {
  "context": <animal
    carnivore="yes"
    sex="female"
  >
    
					
    <species>
      lion
    </species>
    
					
    <weight>
      200
    </weight>
    
					
    <age>
      10
    </age>
    
				
  </animal>,
  "results": Array [
    "An adult lion",
    null,
    null,
  ],
}
`;

exports[`1. Simplest Schematron validation getSchematronResults 2`] = `
Object {
  "context": <animal
    carnivore="no"
    sex="male"
  >
    
					
    <species>
      elephant
    </species>
    
					
    <weight>
      3000
    </weight>
    
					
    <age>
      21
    </age>
    
				
  </animal>,
  "results": Array [
    null,
    "An adult elephant",
    null,
  ],
}
`;
